### PR TITLE
Implemented authentication check and redirect after session expires #111

### PR DESCRIPTION
**Pull request by oliverluethi**

### CODE REVIEW

File: src/biz.dfch.CS.Appclusive.UI/Controllers/LoginController.cs [hunk 1]
```diff
@@ -83,17 +101,58 @@ public ActionResult Logout(string returnUrl = null)
...
+                    SaveChangesDefaultOptions = SaveChangesOptions.PatchOnUpdate,
+                    MergeOption = MergeOption.PreserveChanges,
+                    TenantID = PermissionDecisions.Current.Tenant.Id.ToString(),
+                    Credentials = apiCreds
+                };
+
+                coreRepository.Format.UseJson();
+
+                coreRepository.Tenants.FirstOrDefault();
+
```
Comment (oliverluethi) [inline src/biz.dfch.CS.Appclusive.UI/Controllers/LoginController.cs]: @rufer7 There is no login method exposed from the API (makes sense to me). I've decided to go with a tenant query to check if the credentials work. From my point of view, there is always at least one tenant to query, and every user should have the permission for the root tenant. Can you confirm?

Comment (rufer7) [inline src/biz.dfch.CS.Appclusive.UI/Controllers/LoginController.cs]: @oliverluethi for credential verification purposes I would rather do the following call:

`POST` `~/Diagnostics/Endpoints/AuthenticatedPing`

Example call in PowerShell
```POSH
$svc.diagnostics.InvokeEntitySetActionWithVoidResult("Endpoints", "AuthenticatedPing", $null)
```

File: src/biz.dfch.CS.Appclusive.UI/biz.dfch.CS.Appclusive.UI.csproj [hunk 2]
```diff
@@ -122,10 +126,6 @@
       <Private>True</Private>
       <HintPath>..\packages\Microsoft.AspNet.WebPages.3.2.3\lib\net45\System.Web.Helpers.dll</HintPath>
     </Reference>
-    <Reference Include="System.Web.Mvc, Version=5.2.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <Private>True</Private>
-      <HintPath>..\packages\Microsoft.AspNet.Mvc.5.2.3\lib\net45\System.Web.Mvc.dll</HintPath>
-    </Reference>
```
Comment (rufer7) [inline src/biz.dfch.CS.Appclusive.UI/biz.dfch.CS.Appclusive.UI.csproj]: @oliverluethi why did you change the version of `System.Web.Mvc`? Can you please revert that change?

File: src/biz.dfch.CS.Appclusive.UI/Controllers/LoginController.cs (hunk 1 above)
Comment (rufer7) [inline src/biz.dfch.CS.Appclusive.UI/Controllers/LoginController.cs]: @oliverluethi for credential verification purposes I would rather do the following call:

`POST` `~/Diagnostics/Endpoints/AuthenticatedPing`

Example call in PowerShell
```POSH
$svc.diagnostics.InvokeEntitySetActionWithVoidResult("Endpoints", "AuthenticatedPing", $null)
```

File: src/biz.dfch.CS.Appclusive.UI/biz.dfch.CS.Appclusive.UI.csproj (hunk 2 above)
Comment (oliverluethi) [inline src/biz.dfch.CS.Appclusive.UI/biz.dfch.CS.Appclusive.UI.csproj]: @rufer7 I restored the packages for the project and the System.Web.Mvc DLL from the UI Project was missing. The test project references the System.Web.Mvc.dll from the \packages\Microsoft.Asp.Net.Mvc.5.2.3 folder, but the UI project references a different dll (maybe from a Visual Studio 2015 installation?). 

All i did, was to reference the same DLL as the test project (NuGet).

Comment (rufer7) [inline src/biz.dfch.CS.Appclusive.UI/biz.dfch.CS.Appclusive.UI.csproj]: @oliverluethi hmm strange. Did you run the build and then the package was missing? Strange thing is, that the added line points to version 4.0.0.1. I would still suggest to revert this change as the project builds without problems as well on the build server...

Comment (rufer7) [inline src/biz.dfch.CS.Appclusive.UI/biz.dfch.CS.Appclusive.UI.csproj]: @oliverluethi already changed it so that I can merge the pull request. Thanks!
